compiler/rustc_lint/messages.ftl

```diff
@@ -193,11 +193,10 @@ lint_confusable_identifier_pair = found both `{$existing_sym}` and `{$sym}` as i
     .current_use = this identifier can be confused with `{$existing_sym}`
     .other_use = other identifier used here
 
-lint_dangling_pointers_from_locals = a dangling pointer will be produced because the local variable `{$local_var_name}` will be dropped
-    .ret_ty = return type of the {$fn_kind} is `{$ret_ty}`
-    .local_var = `{$local_var_name}` is part the {$fn_kind} and will be dropped at the end of the {$fn_kind}
+lint_dangling_pointers_from_locals = {$fn_kind} returns a dangling pointer to a dropped local variable
```

Comment:
I can see where this is coming from, but this proposed is less descriptive about the issue. Would it be possible to still mention the name of the local variable?

It is quite useful to be able to quickly see the offending item in the main message, in particular with LSP, where we often don't have the full output of rustc.

Btw, the [`dangling_pointers_from_temporaries` lint](https://doc.rust-lang.org/nightly/rustc/lints/listing/warn-by-default.html#dangling-pointers-from-temporaries) use a very similar warning phrasing. If we change this one, we should also change the other one for consistency.

Reply:
Once you think this is good, I'm happy to take another look at the sibling lint.

Reply:
I restored mention of the name of the local variable in the top level error.